flink-table/flink-table-planner/src/test/java/org/apache/flink/table/test/WithTableEnvironment.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.test;
+
+import org.apache.flink.api.common.RuntimeExecutionMode;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Using this annotation you can inject in the test method:
+ *
+ * <ul>
+ *   <li>{@link TableEnvironment}
+ *   <li>{@link StreamExecutionEnvironment} (Java or Scala)
+ *   <li>{@link StreamTableEnvironment} (Java or Scala)
+ * </ul>
+ *
+ * <p>The underlying parameter injector will infer automatically the type to use from the signature
```

Comment:
Can we add a complete example in Java and Scala here? It will help in the adoption. One example with global class and one with parameters for environment.

Reply:
Not sure I can add it in scala, as the syntax highlight won't work... Anyway, I included it in Java, is it ok?